Px.Utils/Validation/PxFileValidator.cs

```diff
@@ -39,99 +37,159 @@ public void SetCustomValidatorFunctions(CustomSyntaxValidationFunctions? customS
         /// Set custom validators to be used during validation.
         /// </summary>
-        /// <param name="customValidators">Array of objects that implement <see cref="IPxFileValidator"/> interface, used for blocking validation.</param>
-        /// <param name="customAsyncValidators">Array of objects that implement <see cref="IPxFileValidatorAsync"/> interface, used for asynchronous validation.</param>
-        public void SetCustomValidators(IPxFileValidator[]? customValidators = null, IPxFileValidatorAsync[]? customAsyncValidators = null)
+        /// <param name="customStreamValidators">Array of objects that implement <see cref="IPxFileStreamValidator"/> interface, used for px file stream blocking validation.</param>
+        /// <param name="customStreamAsyncValidators">Array of objects that implement <see cref="IPxFileStreamValidatorAsync"/> interface, used for px file stream asynchronous validation.</param>
+        /// <param name="customValidators">Array of objects that implement <see cref="IValidator"/> interface, used for custom blocking validation.</param>
+        /// <param name="customAsyncValidators">Array of objects that implement <see cref="IValidatorAsync"/> interface, used for custom asynchronous validation.</param>
+        public void SetCustomValidators(
+            IPxFileStreamValidator[]? customStreamValidators = null,
+            IPxFileStreamValidatorAsync[]? customStreamAsyncValidators = null,
+            IValidator[]? customValidators = null,
+            IValidatorAsync[]? customAsyncValidators = null)
         {
+            _customStreamValidators = customStreamValidators;
+            _customStreamAsyncValidators = customStreamAsyncValidators;
             _customValidators = customValidators;
             _customAsyncValidators = customAsyncValidators;
         }
 
         /// <summary>
         /// Validates the Px file. Starts with metadata syntax validation, then metadata content validation, and finally data validation.
         /// If any custom validation functions are set, they are executed after the default validation steps.
+        /// <param name="stream">Px file stream to be validated</param>
+        /// <param name="filename">Name of the file subject to validation</param>
+        /// <param name="encoding">Encoding format of the px file</param>
+        /// <param name="fileSystem">File system to use for file operations. If not provided, default file system is used.</param>
+        /// <param name="leaveStreamOpen">Boolean value that determines whether the stream should be left open after validation.
         /// </summary>
         /// <returns><see cref="ValidationResult"/> object that contains the feedback gathered during the validation process.</returns>
-        public ValidationResult Validate()
+        public ValidationResult Validate(
+            Stream stream,
+            string filename,
+            Encoding? encoding = null,
+            bool leaveStreamOpen = false,
```

Comment:
leaveStreamOpen parameters should be removed. Now that the stream is passed as method parameter closing and disposing it the responsibility of the caller.